filesystem/android/src/main/java/com/capacitorjs/plugins/filesystem/FilesystemPlugin.java

```diff
@@ -0,0 +1,494 @@
+package com.capacitorjs.plugins.filesystem;
+
+import android.Manifest;
+import android.content.pm.PackageManager;
+import android.media.MediaScannerConnection;
+import android.net.Uri;
+import com.capacitorjs.plugins.filesystem.exceptions.CopyFailedException;
+import com.capacitorjs.plugins.filesystem.exceptions.DirectoryExistsException;
+import com.capacitorjs.plugins.filesystem.exceptions.DirectoryNotFoundException;
+import com.getcapacitor.JSArray;
+import com.getcapacitor.JSObject;
+import com.getcapacitor.Logger;
+import com.getcapacitor.Plugin;
+import com.getcapacitor.PluginCall;
+import com.getcapacitor.PluginMethod;
+import com.getcapacitor.PluginRequestCodes;
+import com.getcapacitor.annotation.CapacitorPlugin;
+import com.getcapacitor.annotation.Permission;
+import java.io.*;
+import java.nio.charset.Charset;
+import org.json.JSONException;
+
+@CapacitorPlugin(
+    name = "Filesystem",
+    requestCodes = {
+        PluginRequestCodes.FILESYSTEM_REQUEST_WRITE_FILE_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_WRITE_FOLDER_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_READ_FILE_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_READ_FOLDER_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_DELETE_FILE_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_DELETE_FOLDER_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_URI_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_STAT_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_RENAME_PERMISSIONS,
+        PluginRequestCodes.FILESYSTEM_REQUEST_COPY_PERMISSIONS
+    },
+    permissions = {
+        @Permission(permission = Manifest.permission.READ_EXTERNAL_STORAGE, alias = "publicStorage"),
+        @Permission(permission = Manifest.permission.WRITE_EXTERNAL_STORAGE, alias = "publicStorage")
```

Comment:
This can be defined in a cleaner way now as of todays update: https://github.com/ionic-team/capacitor/pull/3822

```suggestion
        @Permission(permission = { Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE }, alias = "publicStorage")
```